### PR TITLE
In-App Review Modal

### LINKED ISSUE #216: Show in-app review modal
Google Play store supports in-app reviews through which an app can request for review and rating from within the app itself [referece](https://developer.android.com/guide/playcore/in-app-review). We want to show an in-app review once a note is saved. 

Flutter has a package to show the in-app review popup: https://pub.dev/packages/in_app_review. 

Showing the popup is as simple as calling this function

```
  Future<void> showReviewPopup() async {
    if (await inAppReview.isAvailable()) {
      await inAppReview.requestReview();
    }
  }
  ```

This is the code which exits the note page after saving, we can show in-app review modal here before exiting. 
```
 else if (state is NoteSavedSuccesfully) {
                showToast(state.newNote!
                    ? S.current.noteSavedSuccessfully
                    : S.current.noteUpdatedSuccessfully);
                _routeToHome();
              }
```

Relevant widgets
```
- lib/features/notes/presentation/pages/note_create_page.dart
-

### PR DESCRIPTION
**Pull request by Prakharpan-dey**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Added code to exit the note page after saving. The in-app review modal is shown before exiting. Additionally, a log file was added to capture any errors while fetching the app review popup.

## Features Covered in this PR

- [x] Added In-App Review Popup

## Related Tickets & Documents

- Related Issue #216
- Closes #216 

## Tested Feature

- Was unable to test it as it requires the app to be published on the Playstore to work.

### CODE REVIEW

**Comment by SankethBK** [comment]
Thanks for contributing!

**Comment by SankethBK** [comment]
There are some formatting issues, please run `dart format --line-length 80 --set-exit-if-changed --output none .` once

File: lib/features/notes/presentation/pages/note_create_page.dart [hunk 1]
```diff
@@ -38,6 +40,8 @@ class _NoteCreatePageState extends State<NoteCreatePage> {
   late double topPadding = 0;
   Timer? _saveTimer;
 
+  final Logger _logger = Logger(); // Initialize the logger
+
```
Comment (SankethBK) [inline lib/features/notes/presentation/pages/note_create_page.dart:44]: use 
```
  final log = printer("NoteCreatePage");
```

**Review by SankethBK — CHANGES_REQUESTED** [review]

File: lib/features/notes/presentation/pages/note_create_page.dart (hunk 1 above)
Comment (Prakharpan-dey) [reply lib/features/notes/presentation/pages/note_create_page.dart:44]: i think the printer function doesnt work like that . Its something like this
 var logger = Logger(
  filter: null, // Use the default LogFilter (-> only log in debug mode)
  printer: PrettyPrinter(), // Use the PrettyPrinter to format and print log
  output: null, // Use the default LogOutput (-> send everything to console)
);
Please correct me if i am wrong

Comment (SankethBK) [reply lib/features/notes/presentation/pages/note_create_page.dart:53]: Hey, `printer` function in `lib/core/logger/logger.dart` is already meant to do same thing, no need to redefine it again

Comment (Prakharpan-dey) [reply lib/features/notes/presentation/pages/note_create_page.dart:44]: I just changes it . Could you please check it?

**Review by SankethBK — APPROVED** [review]

**Pull request merged**